DailyTracker_UI/iOS/Code/Features/Main/Views/Support Views/NavigationButtonView.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  NavigationButtonView.swift
+//  DailyTracker_UI
+//
+//  Created by Mark Golubev on 29/10/2023.
+//
+
+import SwiftUI
+
+struct NavigationButtonView: View {
+    
+    let navigationButton: NavigationButton?
+    
+    var body: some View {
+        navigationButtonView
+    }
+    
+    init(navigationButton: NavigationButton? = NavigationButton(size: 25)) {
+        self.navigationButton = navigationButton
+    }
+    
+    // MARK: Primary Button View
+    @ViewBuilder
+    private var navigationButtonView: some View {
+        if let button = navigationButton {
+            NavigationButton(size: button.size) {
+                // animation here
+                
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.3) {
```

Comment:
is the use of DispatchQueue needed here?

Reply:
no, I'll remove it